### ISSUE TITLE
SwarmUI installed on Linux, but ComfyUI backend wont open, plus missing packages 

### ISSUE THREAD

**Issue by 240db** [mcmonkeyprojects/SwarmUI#306]
### Your question

Hi
I was looking into this project as an alternative to the node based interface. Tried looking for my issue, seems like a lot of people had issues with ComfyUI but they are all variations. 

I was able to install it, following the official instructions in the README.md. It opens the web app with LAN access, all good but the tab  ComfyUI Workflow shows the following:

Failed to load ComfyUI Workflow backend. The server may still be loading. 

The warning ''Backends are still loading on the server" is permanent. I tried turning restarting it several times. 

I tried a few different things

### Logs

When i got to the tab Server , under Logs:

2024-10-05 01:32:20.669 [Init] Program is running.
2024-10-05 01:32:21.170 [Init] Launch web browser...
2024-10-05 01:32:30.889 [Init] Swarm is up to date! You have version 0.9.2.3, and 0.9.2-Beta is the latest.
2024-10-05 01:32:47.081 [Init] Self-Start ComfyUI-0 on port 7821 is loading...
2024-10-05 01:32:47.139 [Warning] [ComfyUI-0/STDERR] Traceback (most recent call last):
2024-10-05 01:32:47.139 [Warning] [ComfyUI-0/STDERR]   File "/home/user/visual_diffuse/SwarmUI/dlbackend/ComfyUI/main.py", line 66, in <module>
2024-10-05 01:32:47.139 [Warning] [ComfyUI-0/STDERR]     import utils.extra_config
2024-10-05 01:32:47.139 [Warning] [ComfyUI-0/STDERR]   File "/home/user/visual_diffuse/SwarmUI/dlbackend/ComfyUI/utils/extra_config.py", line 2, in <module>
2024-10-05 01:32:47.139 [Warning] [ComfyUI-0/STDERR]     import yaml
2024-10-05 01:32:47.139 [Warning] [ComfyUI-0/STDERR] ModuleNotFoundError: No module named 'yaml'
2024-10-05 01:32:47.148 [Info] Self-Start ComfyUI-0 unexpectedly exited (if something failed, change setting `LogLevel` to `Debug` to see why!)
2024-10-05 01:32:47.148 [Info] Self-Start ComfyUI-0 had errors before shutdown:
[ComfyUI-0/STDERR] Traceback (most recent call last):
[ComfyUI-0/STDERR]   File "/home/user/visual_diffuse/SwarmUI/dlbackend/ComfyUI/main.py", line 66, in <module>
[ComfyUI-0/STDERR]     import utils.extra_config
[ComfyUI-0/STDERR]   File "/home/user/visual_diffuse/SwarmUI/dlbackend/ComfyUI/utils/extra_config.py", line 2, in <module>
[ComfyUI-0/STDERR]     import yaml
[ComfyUI-0/STDERR] ModuleNotFoundError: No module named 'yaml'
2024-10-05 01:32:47.972 [Info] Self-Start ComfyUI (Install kornia) exited expectedly but with unexpected exit code 1
2024-10-05 01:32:47.987 [Info] Self-Start ComfyUI (Install sentencepiece) exited expectedly but with unexpected exit code 1
2024-10-05 01:32:47.998 [Info] Self-Start ComfyUI (Install spandrel) exited expectedly but with unexpected exit code 1
2024-10-05 01:32:48.012 [Info] Self-Start ComfyUI (Install rembg) exited expectedly but with unexpected exit code 1
2024-10-05 01:32:48.025 [Info] Self-Start ComfyUI (Install matplotlib==3.9) exited expectedly but with unexpected exit code 1
2024-10-05 01:32:48.039 [Info] Self-Start ComfyUI (Install opencv-python-headless) exited expectedly but with unexpected exit code 1
2024-10-05 01:32:48.053 [Info] Self-Start ComfyUI (Install imageio-ffmpeg) exited expectedly but with unexpected exit code 1
2024-10-05 01:32:48.060 [Info] Self-Start ComfyUI (Install dill) exited expectedly but with unexpected exit code 1
2024-10-05 01:32:48.074 [Info] Self-Start ComfyUI (Install ultralytics==8.1.47) exited expectedly but with unexpected exit code 1

it's really strange because if i open python under the same venv, i can do :
```python
import yaml
```

But not when I run the launch script for linux? Is the script doing anything like activating its own virtual environment? I know Automatic1111 did that in the launch script , it would activate its own virtual environment. Either way, i followed the instructions and searched for similar problems in both SwarmUI and ComfyUI issues posts, both open and closed, the similar issues had no responses. Im considering deleting the whole thing and trying to install again but before that if anyone has a suggestion it would be greatly appreciated!

 

### Other

When I installed it, selected both versions of Flux,I  don't know if that caused an issue but its not obvious from the logs.

**Comment by kalebbroo** [comment]
Can you change your loglevel to debug in the server logs tab and upload your logs using the button on that page? Then provide the link here or in Discord?

**Comment by 240db** [comment]
Hey thanks for the quick response! 

Yeah sure. my bad.

https://paste.denizenscript.com/View/127085

**Comment by 240db** [comment]
Hey just an update! 

I ended up deleting the SwarmUI folder with :
```bash
rm -rf SwarmUI
```

and then just ran install-linux.sh again and installed one model, the Flux1-schnell-fp8. Now the backend is finally loaded after a few seconds, the  ComfyUI Workflow tab opens fine. Im about to run a few tests but its all good! 

**Possible issue:**
First time I ran, I was caught up unexpectedly because i thought I could setup the install remotely through LAN, so what I did was when it asked to go to the localhost page, i stopped the install script after it installed dot-net and etc, tried to find ways to configure it to 0.0.0.0 but then ended up just doing the setup locally by running launch script. Maybe it was because I changed folders from the install then the web install (completing the install with launch-linux.sh because the web app was not installed yet), because i change folders, to SwarmUI to runthe launch script, the install script was one folder above. 

Anyway after I reinstalled it worked, maybe choosing two models can be bad, but not as bad as interrupting the install script.
